### PR TITLE
Bluetooth: controller: Fix to use max. tx power in DTM test mode

### LINKED ISSUE #7243: BLE DTM ll_test does not set correct TXPower
According to the Bluetooth Core Spec v5 Direct Test Mode shall transmit with maximum txpower. However ll_test sets the txpower to 0dBm, which is not the highest txpower for many device including Nordics nRF52 family.

### PR DESCRIPTION
**Pull request by cvinayak**
According to the Bluetooth Specification v5.0, Direct Test
Mode shall use maximum Tx power.

Fixed by adding nRF5x Radio HAL interfaces to get supported
maximum Tx power for SoC selected, and DTM testing sets the
Radio peripheral to use maximum Tx power.

Fixes #7243

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Comment by cvinayak** [comment]
@paulgavrikov please review

**Comment by paulgavrikov** [comment]
@cvinayak looks good

**Comment by cvinayak** [comment]
@paulgavrikov Thank you. You managed to test it? if so, we are good to merge it once others approve it.

**Comment by paulgavrikov** [comment]
@cvinayak Yep, tested on nrf52_pca10040. Both energy consumption and register values are set correctly.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**
